test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -1042,9 +1042,9 @@ TEST_P(ExtProcIntegrationTest, SetHostHeaderRoutingFailed) {
         return true;
       });
 
-  // The routing to upstream is expected to fail and 500 is returned to downstream client, since no
+  // The routing to upstream is expected to fail and 504 is returned to downstream client, since no
   // route is found for mismatched vhost.
-  verifyDownstreamResponse(*response, 500);
+  verifyDownstreamResponse(*response, 504);
```

Comment:
This is a bit surprising, is it actually expected to be a timeout error? I'd expect a 404.

Reply:
Good observation! This is due to test issue and should not be affected by your test change.

500 is returned because the onMessageTimeout is invoked by mistake, which is because test is missing skip_header_processing_on_response (or ext_proc test server should handle request)

I have https://github.com/envoyproxy/envoy/pull/35056 to fix this

Reply:
Thanks for the quick fix! :)